src/gl/index_buffer.js

```diff
@@ -35,12 +39,17 @@ class IndexBuffer {
 
     updateData(array: StructArray) {
         const gl = this.context.gl;
-        assert(this.dynamicDraw);
         // The right VAO will get this buffer re-bound later in VertexArrayObject#bind
         // See https://github.com/mapbox/mapbox-gl-js/issues/5620
         this.context.unbindVAO();
         this.bind();
-        gl.bufferSubData(gl.ELEMENT_ARRAY_BUFFER, 0, array.arrayBuffer);
+        if (this.length === array.length) {
+            assert(this.dynamicDraw || this.resizable);
+            gl.bufferSubData(gl.ELEMENT_ARRAY_BUFFER, 0, array.arrayBuffer);
+        } else {
+            assert(this.resizable);
```

Comment:
Is the resizable boolean really necessary, I assume it is so we can be more explicit about when we want to pay the performance cost of doing a full bufferData.
But could we also do that by adding a warning thats stripped away in production?

Reply:
Is there any advantage to having a buffer non-resizable? Why aren't all buffers resizable by default? We don't necessarily need to resize them.